### PR TITLE
Add levels to story map

### PR DESCRIPTION
**Pull request by IhsenBouallegue**
This PR adds:
- multiple levels to the story map (adding is supported, but editing and deleting not yet)
- editing to all cards

### CODE REVIEW

File: project-canvas/src/components/StoryMapView/AddLevel.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  setLevels,
+  setCases,
+}: {
+  setLevels: Updater<SubActionLevel[]>
+  setCases: Updater<Case[]>
+}) {
+  return (
+    <Button
+      onClick={() => {
+        const levelId = `level-${getRndInteger()}`
```
Comment (bic-marz) [inline project-canvas/src/components/StoryMapView/AddLevel.tsx:16]: probably i am mistaken and i know the chances are eventually very low, but could this not lead to duplicated IDs?

File: project-canvas/src/components/StoryMapView/CaseSubActionLevel.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                subActions={subActions}
+                addSubAction={addSubAction}
+                updateSubAction={updateSubAction}
+              />
+            ))}
+
+          <StrictModeDroppable key="add" droppableId="add" isDropDisabled>
+            {(provided) => (
+              <Stack
+                w="162px"
```
Comment (bic-marz) [inline project-canvas/src/components/StoryMapView/CaseSubActionLevel.tsx:37]: (just flagging this) might lead to mistakes if other values are changed in the future

**Review by bic-marz — APPROVED** [review]
looks good to me, the only real issue i noticed while testing the ui is the absence of an overflow to the right, but it will come with the next features.

File: project-canvas/src/components/StoryMapView/AddLevel.tsx (hunk 1 above)
Comment (IhsenBouallegue) [reply project-canvas/src/components/StoryMapView/AddLevel.tsx:16]: It definitely could. That is a placeholder for now, until the structure is finalized and then will implement a proper ID system

File: project-canvas/src/components/StoryMapView/CaseSubActionLevel.tsx (hunk 2 above)
Comment (IhsenBouallegue) [reply project-canvas/src/components/StoryMapView/CaseSubActionLevel.tsx:37]: Will look into removing that in the next PR

**Pull request merged**